### PR TITLE
Update deps

### PR DESCRIPTION
**Pull request by mattdonnelly**
Fixes a vulnerability in eslint which is added by gulp-eslint. Once you update gulp-eslint though, then you need to update gulp, then you need to update the other gulp plugins etc. etc. Long story most of the packages are on the latest version now 😄

### ESLint Vulnerability Details 📖
CVE: WS-2018-0592

Vulnerable versions: < 4.18.2

Patched Version: 4.18.2

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -159,23 +155,19 @@
...
-    "semi": 2,
-    "space-return-throw-case": 2
+    "semi": 2
   },
   "env": {
     "node": true,
     "mocha": true,
     "es6": true
-  },
-  "ecmaFeatures": {
```
Comment (mattdonnelly) [inline .eslintrc]: Deprecated

File: .eslintrc [hunk 2]
```diff
@@ -29,10 +29,7 @@
         "named": "never"
       }
     ],
-    "space-after-keywords": [
```
Comment (mattdonnelly) [inline .eslintrc]: These changes and the ones below have been renamed in newer eslint versions

File: .babelrc [hunk 3]
```diff
@@ -1,3 +1,3 @@
 {
-  "presets": ["es2015", "stage-1"]
+  "presets": ["@babel/preset-env"]
```
Comment (mattdonnelly) [inline .babelrc]: https://babeljs.io/docs/en/env/

File: gulpfile.js [hunk 4]
```diff
@@ -6,44 +6,42 @@ var istanbul = require('gulp-istanbul');
...
-    .pipe(plumber())
-    .pipe(mocha({reporter: 'spec'}))
-    .on('error', function (err) {
-      mochaErr = err;
-    })
-    .on('end', function () {
-      cb(mochaErr);
-    });
-});
+gulp.task(
```
Comment (mattdonnelly) [inline gulpfile.js]: https://medium.com/@defrian.yarfi/https-medium-com-defrian-yarfi-upgrade-gulp-3-to-gulp-4-the-gulpfile-js-workflow-45196e333b86

File: lib/index.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
-export Client from './client';
```
Comment (mattdonnelly) [inline lib/index.js]: This was an experimental feature that babel no longer includes by default

**Comment by estraph** [comment]
> Long story most of the packages are on the latest version now

Nice one!

**Pull request merged**